src/ILLink.RoslynAnalyzer/OperationAnalysisContextExtensions.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace ILLink.RoslynAnalyzer
+{
+	internal static class OperationAnalysisContextExtensions
+	{
+		/// <summary>
+		/// Finds the symbol of the caller to the current operation, helps to find out the symbol in cases where the operation passes
+		/// through a lambda or a local function.
+		/// </summary>
+		/// <param name="operationContext">Analyzer operation context to retrieve the current operation.</param>
+		/// <param name="targets">Scope of the attribute to search for callers.</param>
+		/// <returns>The symbol of the caller to the operation</returns>
+		public static ISymbol GetContainingSymbol (this OperationAnalysisContext operationContext, DiagnosticTargets targets)
```

Comment:
Was this moved because it's being used in more places?

Reply:
I moved it since it feels like a method that is useful more scenarios other than `Requires...`. Both this and `DiagnosticTargets` can be kept inside `RequiresAnalyzerBase` if we want -- we'd just need to make this function visible to derived classes.